master/internal/core.go

```diff
@@ -451,10 +451,8 @@ func (m *Master) Run(ctx context.Context) error {
 
 	// Restore non-terminal experiments from the database.
 	// Limit the number of concurrent restores at any time within the system to maxConcurrentRestores.
-	// This avoids an issue where the connection pool can be exhausted and the system deadlock. For
-	// example if there are N connections in the pool and N experiments restoring, since a connection
-	// is pinned per restore streaming events, if during any restore blocks for a connection before
-	// completing, we are deadlocked.
+	// This has avoided resource exhaustion in the past (on the db connection pool) and probably is
+	// good still to avoid overwhelming us on restart after a crash.
```

Comment:
> Why not just restore synchronously?

Was more of a "i know it works this way and this isn't what i'm here to fix" rather than i think this is good.

Reply:
I can accept that, but will you make a ticket for the followup work?